x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/SubTests.java

```diff
@@ -71,9 +81,32 @@ public static Iterable<Object[]> parameters() {
                 "SubUnsignedLongsEvaluator[lhs=Attribute[channel=0], rhs=Attribute[channel=1]]",
                 equalTo(asLongUnsigned(lhsBI.subtract(rhsBI).longValue()))
             );
-          })
-          */
-        ));
+          }) */, new TestCaseSupplier("Datetime - Period", () -> {
+            long lhs = (Long) randomLiteral(DataTypes.DATETIME).value();
+            Period rhs = (Period) randomLiteral(EsqlDataTypes.DATE_PERIOD).value();
+            return new TestCase(
+                List.of(new TypedData(lhs, DataTypes.DATETIME, "lhs"), new TypedData(rhs, EsqlDataTypes.DATE_PERIOD, "rhs")),
+                "SubDatetimesEvaluator[lhs=Attribute[channel=0], rhs=Attribute[channel=1]]",
+                DataTypes.DATETIME,
+                equalTo(asMillis(asDateTime(lhs).minus(rhs)))
```

Comment:
We would want that, but the `TestCaseSupplier`s are built statically and the `expectedValue()` is inherited/overridden; otherwise we could have used the `resultMatcher()` method directly (which would have been even nicer).